classes/article/AuthorDAO.inc.php

```diff
@@ -24,38 +24,35 @@ class AuthorDAO extends PKPAuthorDAO {
 	 * Retrieve all published submissions associated with authors with
 	 * the given first name, middle name, last name, affiliation, and country.
 	 * @param $journalId int (null if no restriction desired)
-	 * @param $firstName string
-	 * @param $middleName string
-	 * @param $lastName string
+	 * @param $givenName string
+	 * @param $familyName string
 	 * @param $affiliation string
 	 * @param $country string
 	 */
-	function &getPublishedArticlesForAuthor($journalId, $firstName, $middleName, $lastName, $affiliation, $country) {
+	function &getPublishedArticlesForAuthor($journalId, $givenName, $familyName, $affiliation, $country) {
 		$publishedArticles = array();
 		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 		$params = array(
-			'affiliation',
-			$firstName, $middleName, $lastName,
-			$affiliation, $country
+			IDENTITY_SETTING_GIVENNAME, IDENTITY_SETTING_FAMILYNAME, 'affiliation',
+			$givenName, $familyName, $affiliation, $country
 		);
 		if ($journalId !== null) $params[] = (int) $journalId;
-
 		$result = $this->retrieve(
 			'SELECT DISTINCT
 				aa.submission_id
 			FROM	authors aa
 				LEFT JOIN submissions a ON (aa.submission_id = a.submission_id)
-				LEFT JOIN author_settings asl ON (asl.author_id = aa.author_id AND asl.setting_name = ?)
-			WHERE	aa.first_name = ?
-				AND a.status = ' . STATUS_PUBLISHED . '
-				AND (aa.middle_name = ?' . (empty($middleName)?' OR aa.middle_name IS NULL':'') . ')
-				AND aa.last_name = ?
-				AND (asl.setting_value = ?' . (empty($affiliation)?' OR asl.setting_value IS NULL':'') . ')
+				LEFT JOIN author_settings asgs ON (asgs.author_id = aa.author_id AND asgs.setting_name = ?)
+				LEFT JOIN author_settings asfs ON (asfs.author_id = aa.author_id AND asfs.setting_name = ?)
+				LEFT JOIN author_settings asas ON (asas.author_id = aa.author_id AND asas.setting_name = ?)
+				WHERE a.status = ' . STATUS_PUBLISHED . '
+				AND (asgs.setting_value = ?)
+				AND (asfs.setting_value = ?' . (empty($familyName)?' OR asfs.setting_value IS NULL':'') . ')
+				AND (asas.setting_value = ?' . (empty($affiliation)?' OR asas.setting_value IS NULL':'') . ')
```

Comment:
hmmm... yes... I think it was thought for those parameters never to be null :-\ -- Two authors are considered the same if they both have the same given name, family name, affiliation and country. Thus if family name and/or affiliation of one author is empty, it should be the same for the other. But, you are right -- I will have to solve/consider it differently...
But, now I realized I have a general with this function and I will probably have to solve it in the same way as the function getAuthorsAlphabetizedByJournal, that I haven't solved yet, because I was not sure how :-P -- Actually the given name, family name and affiliation should be the same either for all or at least one locale? And there is no default locale I can use for all authors -- the default/fallback locale for an author is the submission locale and different submissions will surely have different locales... 
Thus, let me think about this function again...